modules/web/js/ballerina/tool-palette/tool-palette-view.jsx

```diff
@@ -324,6 +325,68 @@ class ToolPaletteView extends React.Component {
         return data;
     }
 
+    addDyamicTools(defaultTools) {
+        const environment = this.context.environment;
+        if (environment.getPackages()) {
+            let newToolItem, indexToBeAdded;
+            for (const packageDefintion of environment.getPackages()) {
+                // JMS package
+                if (packageDefintion.getName() === 'ballerina.net.jms') {
+                    newToolItem = {
+                        id: 'jms-service',
+                        name: 'JMS Service',
+                        icon: 'service',
+                        title: 'JMS Service',
+                        nodeFactoryMethod: DefaultNodeFactory.createJMSServiceDef,
+                        description: 'JMS container of resources, each of which defines the logic for'
+                    + ' handling one type of request',
+                    };
+                    defaultTools.tools.forEach((elem, index) => {
+                        if (elem.id === 'constructs_seperator') {
+                            indexToBeAdded = index;
+                        }
+                    });
+                    defaultTools.tools.splice(indexToBeAdded, 0, newToolItem);
+                }
+                // FS package
+                else if (packageDefintion.getName() === 'ballerina.net.fs') {
+                    newToolItem = {
+                        id: 'fs-service',
+                        name: 'FS Service',
+                        icon: 'service',
+                        title: 'FS Service',
+                        nodeFactoryMethod: DefaultNodeFactory.createFSServiceDef,
+                        description: ' FS Server Connector can be used to listen to a ' +
+                        'directory in the local file system',
+                    };
+                    defaultTools.tools.forEach((elem, index) => {
```

Comment:
If this is to find the index, we can use findIndex method of ES6.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex

Reply:
Fixed with 7907698